crates/sui-metrics-proxy/src/main.rs

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::Result;
+use clap::Parser;
+use std::net::SocketAddr;
+use sui_metrics_proxy::admin::{app, set_tracing, shutdown_signal};
+use tracing::info;
+
+const GIT_REVISION: &str = {
+    if let Some(revision) = option_env!("GIT_REVISION") {
+        revision
+    } else {
+        let version = git_version::git_version!(
+            args = ["--always", "--dirty", "--exclude", "*"],
+            fallback = ""
+        );
+
+        if version.is_empty() {
+            panic!("unable to query git revision");
+        }
+        version
+    }
+};
+const VERSION: &str = const_str::concat!(env!("CARGO_PKG_VERSION"), "-", GIT_REVISION);
+
+#[derive(Parser, Debug)]
+#[clap(rename_all = "kebab-case")]
+#[clap(name = env!("CARGO_BIN_NAME"))]
+#[clap(version = VERSION)]
+struct Args {
+    #[clap(long, short, help = "Specify address to listen on")]
+    listen_address: SocketAddr,
```

Comment:
Do we want a default or do we want to require the listen address to always be provided?

Reply:
i'd prefer that it always be provided, but a default is fine.  I know ports 8080-8083 are already spoken for.  In our last sync, we emphasized simple/fast, so not deciding on a port met that criteria.  Let me know if you want to use a specific default.

Reply:
That's fine if you want it explicit, we can leave as-is here